### PR TITLE
T13418

### PR DESCRIPTION
**Pull request by mattdangerw**
https://phabricator.endlessm.com/T13418

### CODE REVIEW

File: search-provider/searchProviderV1.js [hunk 1]
```diff
@@ -1,30 +1,250 @@
...
-        this._search_provider.unregister(connection);
+    LaunchSearch: function (terms, timestamp) {
+        let query = terms.join(' ');
+        this._ensure_app_proxy();
+        this._app_proxy.LoadQueryRemote(query, timestamp);
+    },
+});
+
+const GlobalSearchProvider = new Lang.Class({
+    Name: 'EknGlobalSearchProvider',
```
Comment (ptomato) [inline search-provider/searchProviderV1.js]: Should be changed to `EksGlobalSearchProvider` to reflect its new location?

File: search-provider/applicationV1.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const Lang = imports.lang;
```
Comment (ptomato) [inline search-provider/applicationV1.js]: Any particular reason to split this out into its own file?

File: search-provider/searchProviderV1.js [hunk 3]
```diff
@@ -1,30 +1,250 @@
...
+ * A search provider for a single knowledge app, to be used through dbus by the
+ * shell's global search. Requires the app id of the knowledge app it should run
+ * searches for.
+ *
+ * This search provider will activate the actual knowledge app over dbus with a
+ * result or search to display. As such, it can be used from in or outside of
+ * the actual application process.
+ */
+const AppSearchProvider = Lang.Class({
+    Name: 'EknAppSearchProvider',
```
Comment (ptomato) [inline search-provider/searchProviderV1.js]: Same comment as below, `EksAppSearchProvider`?

File: search-provider/applicationV1.js (hunk 2 above)
Comment (mattdangerw) [inline search-provider/applicationV1.js]: Well that is the file that was originally called searchProviderV1.js before the pull request. Could squash them into one file, partially just laziness here. And seeing as I plan to rewrite this to C as soon as this patch set lands probably not a big deal.

Comment (ptomato) [inline search-provider/applicationV1.js]: OK, got it.

**Pull request merged**